model.go

```diff
@@ -53,6 +53,7 @@ type Model interface {
 	// Blocks returns a map of block type to the message associated with that
 	// block.
 	Blocks() map[string]string
+	SetBlocks(map[string]string)
```

Comment:
**suggestion:** Please add godoc (for me the Block concept in model context is very vague, it is an opportunity to give more hints.